inference-engine/src/mkldnn_plugin/nodes/math.cpp

```diff
@@ -210,11 +210,13 @@ class MathImpl: public ExtLayerBase {
                 dst_data[i] = sinhf(src_data[i]);
             });
             break;
-        case Math::SoftPlus:
+        case Math::SoftPlus: {
+            const float large_positive_value = logf(std::numeric_limits<float>::max());
             parallel_for(dataSize, [&](size_t i) {
-                dst_data[i] = logf(expf(src_data[i]) + 1);
+                dst_data[i] = src_data[i] < large_positive_value ? logf(expf(src_data[i]) + 1.f) : src_data[i];
```

Comment:
Perhaps just need to divide by src_data sign?
x >= 0 ? x - ln(exp(-x) + 1) : ln(exp(x) + 1)